app/Main/Cabal.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE ApplicativeDo #-}
+module Main.Cabal where
+
+import           Data.Aeson           (encode)
+import           Data.Bool            (bool)
+import qualified Data.ByteString.Lazy as BL
+import           Data.List.NonEmpty   (NonEmpty (..), nonEmpty)
+import           Data.Maybe           (fromMaybe)
+import           Options.Applicative
+
+import           Graphex.Cabal
+import           Graphex.LookingGlass
+
+data CabalOptions = CabalOptions
+  { optToDiscover      :: [CabalDiscoverType]
+  , optIncludeExternal :: Bool
+  } deriving stock Show
+
+justWhen :: a -> Bool -> Maybe a
+justWhen a True  = Just a
+justWhen _ False = Nothing
+
+cabalOptions :: Parser CabalOptions
+cabalOptions = do
+  optToDiscover <- fmap mconcat $ many $ asum
```

Comment:
This collects the options and preserves order. The actual runner then folds them left-to-right and this type's Semigroup instance does the fun stuff.